sync2jira/downstream_issue.py

```diff
@@ -152,6 +152,68 @@ def execute_snowflake_query(issue):
     return results
 
 
+def _get_field_id_by_name(client, field_name):
+    """
+    Convert a human-readable custom field name to its JIRA field ID.
+
+    :param jira.client.JIRA client: JIRA client
+    :param str field_name: Human-readable field name (e.g., "Story Points", "Epic Link")
+    :returns: Field ID (e.g., "customfield_12310243") or None if not found
+    :rtype: Optional[str]
+    """
+    # Cache field mappings per client to avoid repeated API calls
+    cache_key = id(client)
+    if cache_key not in jira_cache:
+        jira_cache[cache_key] = {}
+
+    # Check cache first
+    if field_name in jira_cache[cache_key]:
+        return jira_cache[cache_key][field_name]
```

Comment:
This idiom is unsatisfying because it does the `dict` lookup _twice_ (on what is supposed to be the _fast_ path) -- once for the `in` operation and again for the `[]` reference.


I'd recommend something like this:
```suggestion
    field_ID = jira_cache[cache_key].get(field_name)
    if field_ID:
        return field_ID
```

And, I would recommend using the "walrus" operator for this:
```suggestion
    if field_ID := jira_cache[cache_key].get(field_name)
        return field_ID
```